### PR TITLE
feat(console): code editor component

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Use VSCode's editor "Monaco Editor" as our code editor and code preview with hightlight.

Monaco editor is loaded on demand, won't increse the bundle size:

before:
截屏2022-03-11 下午3 20 29

after:
截屏2022-03-11 下午3 20 11

## Linear Issue Reference

LOG-1708

## Testing

截屏2022-03-11 下午3 15 17

dark mode:
截屏2022-03-11 下午3 15 36

@demonzoo @gao-sun

### CODE REVIEW

File: packages/console/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   "dependencies": {
     "@logto/phrases": "^0.1.0",
     "@logto/schemas": "^0.1.0",
+    "@monaco-editor/react": "^4.3.1",
```
Comment (gao-sun) [inline packages/console/package.json:21]: `monaco-editor` is a peer dependency of this

File: packages/console/src/components/CodeEditor/index.tsx [hunk 2]
```diff
@@ -1,21 +1,50 @@
...
+  height = '300px',
+  isReadonly = false,
+  isDarkMode,
+}: Props) => {
+  const handleChange = (changedValue: string | undefined) => {
+    onChange?.(changedValue ?? '');
+  };
+
+  // See https://microsoft.github.io/monaco-editor/api/enums/monaco.editor.EditorOption.html
+  const options = {
```
Comment (gao-sun) [inline packages/console/src/components/CodeEditor/index.tsx:28]: you can use typed options from `monaco-editor`

**Review by gao-sun — CHANGES_REQUESTED** [review]

**Comment by gao-sun** [comment]
overall looks ok though

File: packages/console/package.json (hunk 1 above)
Comment (wangsijie) [reply packages/console/package.json:21]: updated

File: packages/console/src/components/CodeEditor/index.tsx (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/components/CodeEditor/index.tsx]: updated

**Review by xiaoyijun — APPROVED** [review]
LGTM

File: packages/console/src/components/CodeEditor/index.tsx [hunk 3]
```diff
@@ -1,21 +1,51 @@
...
+  value,
+  onChange,
+  language,
+  height = '300px',
+  isReadonly = false,
+  isDarkMode,
+}: Props) => {
+  const handleChange = (changedValue: string | undefined) => {
+    onChange?.(changedValue ?? '');
+  };
```
Comment (demonzoo) [inline packages/console/src/components/CodeEditor/index.tsx:26]:
```suggestion
  const handleChange = (changedValue = '') => {
    onChange?.(changedValue);
  };
```

File: packages/console/src/components/CodeEditor/index.tsx [hunk 4]
```diff
@@ -1,21 +1,51 @@
...
+
+  // See https://microsoft.github.io/monaco-editor/api/enums/monaco.editor.EditorOption.html
+  const options: monaco.editor.IStandaloneEditorConstructionOptions = {
+    readOnly: isReadonly,
+    scrollBeyondLastLine: false,
+    codeLens: false,
+    minimap: {
+      enabled: false,
+    },
+    folding: false,
```
Comment (demonzoo) [inline packages/console/src/components/CodeEditor/index.tsx:36]: Is it possible to adjust the font and color used in the editor?

**Review by demonzoo — APPROVED** [review]
Overall looking good

**Review by gao-sun — APPROVED** [review]

File: packages/console/src/components/CodeEditor/index.tsx (hunk 4 above)
Comment (wangsijie) [reply packages/console/src/components/CodeEditor/index.tsx:36]: yep, just set some options if needed.

**Pull request merged**